pvlib/irradiance.py

```diff
@@ -3117,3 +3117,72 @@ def complete_irradiance(solar_zenith,
                                      'dhi': dhi,
                                      'dni': dni})
     return component_sum_df
+
+
+def pvl_louche(ghi, solar_zenith, datetime_or_doy):
+    """
+    Determine DNI and GHI from GHI using Louche model.
+
+    Parameters
+    ----------
+    ghi : Series
+        Global horizontal irradiance. [W/m^2]
+
+    solar_zenith : Series
+        True (not refraction-corrected) zenith angles in decimal
+        degrees. Angles must be >=0 and <=90.
+
+    datetime_or_doy : numeric, pandas.DatetimeIndex
+        Day of year or array of days of year e.g.
+        pd.DatetimeIndex.dayofyear, or pd.DatetimeIndex.
+
+    Returns
+    -------
+    data: OrderedDict or DataFrame
+        Contains the following keys/columns:
+
+            * ``dni``: the modeled direct normal irradiance in W/m^2.
+            * ``dhi``: the modeled diffuse horizontal irradiance in
+              W/m^2.
+            * ``kt``: Ratio of global to extraterrestrial irradiance
+              on a horizontal plane.
+    Notes
+    -----
+    `solar_zenith` angles should be between 0 and 90 degrees else NaN will be
+    returned. Day of year (doy) should be between 1 and 366 else NaN will be
+    returned.
+
+    References
+    -------
+    .. [1] Louche A, Notton G, Poggi P, Simmonnot G. Correlations for direct
+       normal and global horizontal irradiation on French Mediterranean site.
+       Solar Energy 1991;46:261-6
+
+    """
+    bool = np.logical_or(solar_zenith > 90, solar_zenith < 0)
+    solar_zenith = np.where(bool, np.NaN, solar_zenith)
+
+    if np.isscalar(datetime_or_doy):
+        bool = (np.any(datetime_or_doy > 366 or datetime_or_doy < 1, axis=0))
+        datetime_or_doy = np.where(bool, np.NaN, datetime_or_doy)
+
+    # this is the I0 calculation from the reference
+    # SSC uses solar constant = 1366.1
```

Comment:
I'm not sure this comment is correct -- should it be deleted?

Reply:
SSC default value in ```get_extra_radiation``` was 1366.1  ,I just thought to mention it explicitly as done in ```disc```.
Also have a look at ```disc``` in comments it mentions 1367 but uses 1370.

Reply:
I can't tell from the reference [1] what solar constant was used. He references another paper for the direct irradiance reference which uses 1367, but it's not clear that 1367 was also used to make the correlation defining `kb`. I think it would be better to drop the comments on lines 3163-3164.